generate_tags.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+import requests
+
+# This script loads the enterprise-attack.json mitre dataset into a dict (mitre_dataset)
+# according to the parameters set below. The dict is then iterated to produce an attack.py
+# file, containing variables for the multiple Mitre ATT&CK tags relevant to the project.
+#
+# The enterprise-attack.json file is sourced from Mitre's repository at
+# https://github.com/mitre-attack/attack-stix-data/tree/master/enterprise-attack.
+
+# Replace the URL below with the version of your choice.
+
+mitre_source = (
+    "https://raw.githubusercontent.com/mitre-attack/attack-stix-data/"
+    "master/enterprise-attack/enterprise-attack-14.1.json"
+)
+
+# Begin parameters; modify these to customize the tags produced by the script
+
+included_platforms = ["Windows", "Linux"]
+
+# End parameters
+
+
+def retrieve_mitre_data(url):
+    print("Attempting download of Mitre data, please wait...")
+    response = requests.get(url)
+    if response.status_code == 200:
+        print("\nDownload finished!")
+    else:
+        print("Error: ", response.status_code)
+    return json.loads(response.content)
+
+
+def reformat_technique(target_technique):
+    #
+    # This function reformats the given technique from the mitre dataset to conform
+    # to the specs of monkeyevents.
+    #
+    # param target_technique dict - technique object from the mitre dataset
+    # returns a specially-formatted string to suit the specs of monkeyevents
+    #
+    return (
+        "_".join(
+            [
+                target_technique["name"].upper().replace(" ", "_"),
+                target_technique["external_references"][0]["external_id"].upper(),
+            ]
+        )
+        .replace("-", "_")
+        .replace(".", "_")
+        .replace("/", "_")
+        .replace("(", "")
+        .replace(")", "")
+        .replace("Ä", "A")
+    )
+
+
+mitre_dataset = retrieve_mitre_data(mitre_source)
+if mitre_dataset:
+    valid_techniques = []
+    for technique in mitre_dataset.get("objects", []):
+        if (
+                technique.get("type") == "attack-pattern" and
+                not technique.get("x_mitre_deprecated") and
+                not technique.get("revoked")
+        ):
+            x_mitre_platforms = technique.get("x_mitre_platforms", [])
+            if any(platform in x_mitre_platforms for platform in included_platforms):
```

Comment:
This is a lot of logic to digest all at once here. If we wrap it in a well-named function, for example `technique_matches_criteria()`, then we can just say
```python
if technique_matches_criteria(technique):
    # format and append
    ...
```

This way, If I want to understand what techniques are included, there's a clearly defined place where I can look. And if I don't care, I don't need to be destracted with those details while I'm reading the rest of the code.

NOTE: Naming is hard. `technique_matches_criteria` may not be the ideal name. The goal is to have something that reads as close to plain English as possible. `if technique_matches_criteria()` is pretty easy for anyone to understand because it's basically just english. Feel free to suggest another name.